商品出品機能の実装

kumazakiz opened the pull request:
#WHAT
-商品出品機能の実装。

#WHY
-商品の情報を保存できるようにするため。

-itemモデルの単体テストコード
https://gyazo.com/ec3211cc3c74bc2968dcc6956650bbaa

-商品出品ページのエラーメッセージ表示
https://gyazo.com/cb3fd79e6de6f96b503816dd42c676c3

-商品出品の登録
https://gyazo.com/969f94611ca24bea722e714eea71415a

kumazakiz committed:
itemモデル・テーブルの作成

kumazakiz committed:
ActiveStorageの導入

kumazakiz committed:
itemモデルのルーティング・コントローラーの準備

kumazakiz committed:
userモデルとitemモデルのアソシエーション設定

kumazakiz committed:
ActiveHashの導入

kumazakiz committed:
itemモデルのバリデーションの設定

kumazakiz committed:
商品出品ページ(ActiveHash実装)

kumazakiz committed:
商品出品ページ(保存設定)

kumazakiz committed:
itemモデルの単体テストコードの実装

kumazakiz committed:
JavaScriptの実装(販売価格の表示設定)

kumazakiz committed:
Rubocopコマンドでの修正

mizuki-iio commented on app/controllers/items_controller.rb:
```diff
@@ -1,4 +1,27 @@
+  before_action :authenticate_user!, except: [:index, :new]
+
   def index
+    @item = Item.new
```
こちら商品一覧機能実装の際に記述いただければと思いますので、現段階では削除またはコメントアウトしておきましょう！

mizuki-iio commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,107 @@
+        expect(@item.errors.full_messages).to include('Selling price 300~9,999,999の間で設定してください。')
+      end
+      it 'selling_priceが¥299以下だと登録できない' do
+        @item.selling_price = '299'
```
こちらpriceは数値型になるかと思いますので、シングルクウォートで加工必要はございません！

mizuki-iio commented on spec/models/item_spec.rb:
```diff
@@ -0,0 +1,107 @@
+        expect(@item.errors.full_messages).to include('Selling price 300~9,999,999の間で設定してください。')
+      end
+      it 'selling_priceが¥10,000,000以上だと登録できない' do
+        @item.selling_price = '10000000'
```
こちらpriceは数値型になるかと思いますので、シングルクウォートで加工必要はございません！

mizuki-iio reviewed (commented):
レビュー依頼ありがとうございます！
3点コメントいたしましたので、ご確認をお願いいたします！

kumazakiz committed:
コードレビュー(修正依頼分)

内容
①items_controllerのindexを空にする。
②item_spec.rbのselling_priceの'299'の''を取る。
③item_spec.rbのselling_priceの'10000000'の''を取る。

JUNSHIROBABA commented on app/javascript/selling_price.js:
```diff
@@ -0,0 +1,18 @@
+  const priceInput = document.getElementById("item-price");
+  priceInput.addEventListener("input", () => {
+    const inputValue = priceInput.value;
+    console.log(inputValue);
```
不要な`console.log`はコードレビュー依頼前に削除しておきましょう。
消すことが出来ない場合はコメントアウトしておき、残しておく理由を明記しておきましょう。

**理由**
コードレビューは「一旦完成したコード」に対して行うものです。
console.logやbinding.pryは開発中の検証で使用しているものかと存じます。本番環境にアップロードする前に消しておくことで、予期せぬバグを防止できます。

JUNSHIROBABA commented on app/javascript/selling_price.js:
```diff
@@ -0,0 +1,18 @@
+    const addTaxDom = document.getElementById("add-tax-price");
+    addTaxDom.innerHTML = Math.floor(inputValue/10);
+    const priceTax = Math.floor(inputValue/10);
+    console.log(addTaxDom.textContent);
```
不要な`console.log`はコードレビュー依頼前に削除しておきましょう。
消すことが出来ない場合はコメントアウトしておき、残しておく理由を明記しておきましょう。

**理由**
コードレビューは「一旦完成したコード」に対して行うものです。
console.logやbinding.pryは開発中の検証で使用しているものかと存じます。本番環境にアップロードする前に消しておくことで、予期せぬバグを防止できます。

JUNSHIROBABA commented on app/javascript/selling_price.js:
```diff
@@ -0,0 +1,18 @@
+
+    const profitDom = document.getElementById("profit");
+    profitDom.innerHTML = Math.floor(inputValue - priceTax);
+    console.log(profitDom.textContent);
```
不要な`console.log`はコードレビュー依頼前に削除しておきましょう。
消すことが出来ない場合はコメントアウトしておき、残しておく理由を明記しておきましょう。

**理由**
コードレビューは「一旦完成したコード」に対して行うものです。
console.logやbinding.pryは開発中の検証で使用しているものかと存じます。本番環境にアップロードする前に消しておくことで、予期せぬバグを防止できます。

JUNSHIROBABA commented on spec/factories/categories.rb:
```diff
@@ -0,0 +1,4 @@
+FactoryBot.define do
```
Active Hashで生成されたモデルのFactoryBotは不要かと思いますので、削除しましょう。

JUNSHIROBABA reviewed (commented):
レビュー依頼ありがとうございます。
追加で４点程コメントいたしました。ご確認をお願いいたします。

また、合わせて以下の項目に関するgif動画をプルリクエストへ添付いただけますでしょうか。

````
ログインしていないユーザーが出品しようとするとリダイレクトされる動画
````

お手数をお掛けしますが、よろしくお願いいたします。

kumazakiz committed:
コードレビュー修正依頼(2)

修正内容
①selling_price.jsのconsole.log~の記述を全部削除する。
②items.rbのActiveHashにかかってたFactoryBotの記述を削除する。

mizuki-iio reviewed (commented):
レビュー依頼ありがとうございます！
こちらLGTMとさせていただきます！

実装お疲れ様でした！
引き続き頑張って進めていきましょう！
